diff --git a/www/install/steps/step8.php b/www/install/steps/step8.php
index a37127a..0123d79 100644
--- a/www/install/steps/step8.php
+++ b/www/install/steps/step8.php
@@ -44,15 +44,6 @@ $title = _('Installation finished');
 
 $contents = '<div>'._('Congratulations, you have successfully installed Centreon!').'</div>';
 
-$centreon_path = realpath(dirname(__FILE__) . '/../../../');
-
-if (false === is_dir($centreon_path . '/installDir')) {
-    $contents .= '<br>Warning : The installation directory cannot be move. Please create the directory ' . $centreon_path . '/installDir and give it the rigths to apache user to write.';
-} else {
-    $name = 'install-' . $_SESSION['version'] . '-' . date('Ymd_His');
-    @rename(str_replace('steps', '', getcwd()), $centreon_path . '/installDir/' . $name);
-}
-
 session_destroy();
 require_once ("process/advertising.php");
 $template->assign('step', STEP_NUMBER);
@@ -62,3 +53,11 @@ $template->assign('finish', 1);
 $template->assign('blockPreview', 1);
 $template->display('content.tpl');
 
+$centreon_path = realpath(dirname(__FILE__) . '/../../../');
+
+if (false === is_dir($centreon_path . '/installDir')) {
+    $contents .= '<br>Warning : The installation directory cannot be move. Please create the directory ' . $centreon_path . '/installDir and give it the rigths to apache user to write.';
+} else {
+    $name = 'install-' . $_SESSION['version'] . '-' . date('Ymd_His');
+    system("mv ". realpath(str_replace('steps', '', getcwd())) . " " .  $centreon_path . '/installDir/' . $name);
+}
\ No newline at end of file
