diff --git a/www/install/step_upgrade/step5.php b/www/install/step_upgrade/step5.php
index 75d1b81..bb01290 100644
--- a/www/install/step_upgrade/step5.php
+++ b/www/install/step_upgrade/step5.php
@@ -47,16 +47,6 @@ $contents = sprintf(
     $_SESSION['CURRENT_VERSION']
 );
 
-$centreon_path = realpath(dirname(__FILE__) . '/../../../');
-
-if (false === is_dir($centreon_path . '/installDir')) {
-    $contents .= '<br>Warning : The installation directory cannot be move. Please create the directory '
-        . $centreon_path . '/installDir and give it the rigths to apache user to write.';
-} else {
-    $name = 'install-' . $_SESSION['CURRENT_VERSION'] . '-' . date('Ymd_His');
-    @rename(str_replace('step_upgrade', '', getcwd()), $centreon_path . '/installDir/' . $name);
-}
-
 session_destroy();
 
 $template->assign('step', STEP_NUMBER);
@@ -65,3 +55,13 @@ $template->assign('content', $contents);
 $template->assign('finish', 1);
 $template->assign('blockPreview', 1);
 $template->display('content.tpl');
+
+$centreon_path = realpath(dirname(__FILE__) . '/../../../');
+
+if (false === is_dir($centreon_path . '/installDir')) {
+    $contents .= '<br>Warning : The installation directory cannot be move. Please create the directory '
+        . $centreon_path . '/installDir and give it the rigths to apache user to write.';
+} else {
+    $name = 'install-' . $_SESSION['CURRENT_VERSION'] . '-' . date('Ymd_His');
+    system('mv '. str_replace('step_upgrade', '', getcwd()) . ' ' . $centreon_path . '/installDir/' . $name);
+}
\ No newline at end of file
